src/test/java/com/ning/billing/recurly/TestRecurlyClient.java

```
@@ -1570,4 +1572,34 @@ public void testCounts() throws Exception {
         Integer giftCardsCount = recurlyClient.getGiftCardsCount(qp);
         Assert.assertNotNull(giftCardsCount);
     }
+
+    @Test(groups = "integration")
+    public void testPurchase() throws Exception {
+        final Account accountData = TestUtils.createRandomAccount();
+        final BillingInfo billingInfoData = TestUtils.createRandomBillingInfo();
+        final Adjustments adjustmentsData = new Adjustments();
+        final Adjustment adjustmentData = TestUtils.createRandomAdjustment();
+        adjustmentData.setCurrency(null); // this one accepted by site
+        adjustmentsData.add(adjustmentData);
+
+        billingInfoData.setAccount(null); // null out random account fixture
+        accountData.setBillingInfo(billingInfoData); // add the billing info to account data
+
+        try {
+            final Purchase purchaseData = new Purchase();
+            purchaseData.setAccount(accountData);
+            purchaseData.setAdjustments(adjustmentsData);
+            purchaseData.setCollectionMethod("automatic");
+            purchaseData.setAdjustments(adjustmentsData);
+            purchaseData.setCurrency("USD");
+
+            final Invoice invoice = recurlyClient.purchase(purchaseData);
+            Assert.assertEquals(invoice.getLineItems().get(0).getProductCode(), adjustmentData.getProductCode());
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
```

Sorry, removing. I think this may actually swallow the exception.